src/lib/viewers/image/ImageViewer.js

```diff
@@ -165,6 +187,36 @@ class ImageViewer extends ImageBaseViewer {
         };
     }
 
+    /**
+     * Gets the viewport dimensions.
+     *
+     * @return {Object} the width & height of the viewport
+     */
+    getViewportDimensions() {
+        return {
+            width: this.wrapperEl.clientWidth - 2 * IMAGE_PADDING,
+            height: this.wrapperEl.clientHeight - 2 * IMAGE_PADDING,
+        };
+    }
+
+    /**
+     * Sets mode to be AnnotationMode.NONE if the zoomed image overflows the viewport.
+     *
+     * @return {void}
+     */
+    handleZoomEvent({ newScale }) {
+        const [width, height] = newScale;
+
+        const viewport = this.getViewportDimensions();
```

Comment:
Destructure?

Reply:
I think I can keep it as is because if I were to destructure it there would be a naming collision between the existing width / height defined on L208.

I could do this though:

`const { height: viewportHeight, width: viewportWidth } = this.getViewportDimensions();`

Reply:
Yeah, `viewport.height/width` is only used once so it's probably fine.